src/controller/artifact/annotation/checker/v1alpha1.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/pkg/registry"
+)
+
+const (
+	// V1alpha1 is the version of annotation checker
+	V1alpha1 = "v1alpha1"
+)
+
+// RegisterV1Alpha1Checker register v1alpha1 annotation checker
+func RegisterV1Alpha1Checker() error {
+	return RegisterAnnotationChecker(NewV1Alpha1Checker(), V1alpha1)
+}
+
+// NewV1Alpha1Checker creates a new v1alpha1 checker
+func NewV1Alpha1Checker() *V1Alpha1Checker {
+	return &V1Alpha1Checker{}
+}
+
+// V1Alpha1Checker is an annotation checker whose version is v1alpha1
+type V1Alpha1Checker struct {
+	RegCli registry.Client
+}
+
+// ProcessSkipList process skip-list annotation
+func (p *V1Alpha1Checker) ProcessSkipList(artifact *artifact.Artifact, config map[string]interface{}, skipListKey, skipListValue string) error {
+
+	skipKeyList := strings.Split(skipListValue, ",")
+
+	for _, skipKey := range skipKeyList {
+		delete(config, skipKey)
+	}
+
+	artifact.ExtraAttrs = config
+
+	return nil
+}
+
+// ProcessIcon process icon annotation
+func (p *V1Alpha1Checker) ProcessIcon(artifact *artifact.Artifact, digest string) error {
+	// pull icon layer
+	_, icon, err := p.RegCli.PullBlob(artifact.RepositoryName, digest)
+
+	// check the size of the size <= 1MB
+	data, err := ioutil.ReadAll(io.LimitReader(icon, 1<<20))
+	if err != nil {
+		if err == io.EOF {
+			return fmt.Errorf("the maximum size of the icon is 1MB")
```

Comment:
The 1MB is Harbor's limiation? If so, can we explain it at somewhere? Doc?

Reply:
Process Icon part @ywk253100 helped code this part. 
Maybe it is a limitation we defined? @ywk253100